src/api/authApi.js

```diff
@@ -0,0 +1,65 @@
+
+import router from '@/router'
+import axios from 'axios'
+
+export const URL_KEY = 'AIzaSyDLzMdBM3yeRBVTM3z6fbSh3p8TYOq-0jQ'
+
+export const axiosAuth = axios.create({
+  baseURL: 'https://identitytoolkit.googleapis.com/v1'
+})
+
+axiosAuth.interceptors.request.use(config => {
+  return config
+})
+
+axiosAuth.interceptors.response.use(response => {
+  return response
+}, function (error) {
+  if (error.response.status === 400 && error.response.data.error.message === 'EMAIL_EXISTS') {
+    alert(error.response.data.error.message + ' please log in')
+    router.push('/login')
+    return Promise.reject(error)
+  }
+
+  if (error.response.status === 400 && error.response.data.error.message === 'EMAIL_NOT_FOUND') {
+    alert(error.response.data.error.message + ' please register your email!')
+    router.push('/signup')
+    return Promise.reject(error)
+  }
+
+  if (error.response.status === 400 && error.response.data.error.message === 'WEAK_PASSWORD') {
+    alert(error.response.data.error.message + ' please enter correct password!')
+  }
+
+  if (error.response.status === 400 && error.response.data.error.message === 'INVALID_PASSWORD') {
+    alert(error.response.data.error.message + ' please enter correct password!')
+  }
+
+  console.log('status: ', error.response.status, 'message: ', error.response.data.error.message)
+
+  return Promise.reject(error)
+})
+
+export function authUser (
```

Comment:
This method should not be here. You should have: loginUser / signupUser actions in the store. Without any dependency on this. You just need to import `axiosAuth` inside the store and that's all.